### ISSUE TITLE
PyTorch/XLA Codegen Migration

### ISSUE THREAD

**Issue by wonjoolee95** [pytorch/xla#3560]
## 🚀 Feature
As PyTorch/XLA migrates to the LTC (Lazy Tensor Core), we need to clean up the existing stub code (which spans over 6+ files) that were used to do the op lowering. The complete process and file structure for the old op lowering can be found in [this doc](https://github.com/pytorch/xla/blob/master/OP_LOWERING_GUIDE.md). Replacing the supported op with the codegen SHOULD NOT introduce any new behavior, it is purely for the clean up purpose. 

TODO migrate @JackCaoG's codegen guide doc to a publicly available .md in the repo. 

## Ops

List of ops that we will replace with full codgen. Please add the following information under each op that you choose to work on: the status of the task, link to the PR, and the owner. For example:
```
  - Status: Merged
  - PR: https://github.com/pytorch/xla/pull/3565
  - Owner: @wonjoolee95  
```

Once the corresponding PR is merged, feel free to check the box to mark completion. 

- [ ] \_\_ilshift__.Scalar
- [ ] \_\_ilshift__.Tensor
- [ ] \_\_irshift__.Scalar
- [ ] \_\_irshift__.Tensor
- [ ] \_\_lshift__.Scalar
- [ ] \_\_lshift__.Tensor
- [ ] \_\_rshift__.Scalar
- [ ] \_\_rshift__.Tensor
- [ ] \_adaptive_avg_pool2d
- [ ] \_adaptive_avg_pool2d_backward
- [ ] \_adaptive_avg_pool3d
- [ ] \_adaptive_avg_pool3d_backward
- [ ] \_amp_foreach_non_finite_check_and_unscale_
- [ ] \_amp_update_scale_
- [ ] \_copy_from
- [ ] \_copy_from_and_resize
- [ ] \_index_put_impl_
- [ ] \_local_scalar_dense
- [ ] \_log_softmax
- [ ] \_log_softmax_backward_data
- [ ] \_pack_padded_sequence
- [ ] \_softmax
- [ ] \_softmax_backward_data
- [ ] \_to_cpu
- [ ] \_trilinear
- [ ] \_unsafe_view
- [x] abs
  - Status: Merged
  - PR: https://github.com/pytorch/xla/pull/3544
  - Owner: @JackCaoG 
- [x] acos
  - Status: Merged
  - PR: https://github.com/pytorch/xla/pull/3564
  - Owner: @wonjoolee95  
- [x] acosh
  - Status: Merged
  - PR: https://github.com/pytorch/xla/pull/3564
  - Owner: @wonjoolee95  
- [ ] adaptive_max_pool2d
- [ ] adaptive_max_pool2d_backward
- [ ] add.Scalar
- [ ] add.Tensor
- [ ] addcdiv
- [ ] addcdiv_
- [ ] addcmul
- [ ] addmm
- [ ] alias
- [ ] all
- [ ] all.dim
- [ ] amax
- [ ] amin
- [ ] any
- [ ] any.dim
- [ ] arange.start_out
- [ ] argmax
- [ ] argmin
- [ ] as_strided
- [ ] as_strided_
- [x] asin
  - Status: Merged
  - PR: https://github.com/pytorch/xla/pull/3565
  - Owner: @wonjoolee95  
- [x] asinh
  - Status: Merged
  - PR: https://github.com/pytorch/xla/pull/3565
  - Owner: @wonjoolee95  
- [x] atan
  - Status: Merged
  - PR: https://github.com/pytorch/xla/pull/3565
  - Owner: @wonjoolee95  
- [ ] atan2
- [x] atanh
  - Status: Merged
  - PR: https://github.com/pytorch/xla/pull/3565
  - Owner: @wonjoolee95  
- [ ] avg_pool2d
- [ ] avg_pool2d_backward
- [ ] avg_pool3d
- [ ] avg_pool3d_backward
- [ ] baddbmm
- [ ] bernoulli
- [ ] bernoulli_.float
- [ ] bernoulli_.Tensor
- [ ] binary_cross_entropy
- [ ] binary_cross_entropy_backward
- [ ] binary_cross_entropy_with_logits
- [ ] bitwise_and.Scalar
- [ ] bitwise_and.Tensor
- [ ] bitwise_not.out
- [ ] bitwise_or.Scalar_out
- [ ] bitwise_or.Tensor_out
- [ ] bitwise_xor.Scalar_out
- [ ] bitwise_xor.Tensor_out
- [ ] bmm
- [ ] cat
- [ ] ceil
- [ ] celu
- [ ] celu_
- [ ] cholesky
- [ ] clamp
- [ ] clamp.Tensor
- [ ] clamp_max
- [ ] clamp_max.Tensor_out
- [ ] clamp_min
- [ ] clamp_min.Tensor_out
- [ ] clone
- [ ] constant_pad_nd
- [ ] convolution_backward_overrideable
- [ ] convolution_overrideable
- [x] cos
  - Status: Merged
  - PR: https://github.com/pytorch/xla/pull/3574
  - Owner: @miladm 
- [x] cosh
  - Status: Merged
  - PR: https://github.com/pytorch/xla/pull/3574
  - Owner: @miladm 
- [ ] cross
- [ ] cumprod
- [ ] cumsum
- [ ] diag
- [ ] diagonal
- [ ] div.Scalar
- [ ] div.Tensor
- [ ] div.Tensor_mode
- [ ] dot
- [ ] elu
- [ ] elu_
- [ ] elu_backward
- [ ] embedding
- [ ] embedding_dense_backward
- [ ] empty.memory_format
- [ ] empty_strided
- [ ] eq.Scalar
- [ ] eq.Tensor
- [ ] erf
- [ ] erfc
- [ ] erfinv
- [ ] exp
- [ ] expand
- [ ] expm1
- [ ] exponential_
- [ ] eye.m_out
- [ ] eye.out
- [ ] fill_.Scalar
- [ ] fill_.Tensor
- [ ] flip
- [ ] floor
- [ ] fmod.Scalar
- [ ] fmod.Tensor
- [ ] frac
- [ ] gather
- [ ] ge.Scalar
- [ ] ge.Tensor
- [ ] gelu
- [ ] gelu_backward
- [ ] ger
- [ ] gt.Scalar
- [ ] gt.Tensor
- [ ] hardshrink
- [ ] hardshrink_backward
- [ ] hardsigmoid
- [ ] hardsigmoid_backward
- [ ] hardswish
- [ ] hardswish_backward
- [ ] hardtanh
- [ ] hardtanh_backward
- [ ] index.Tensor
- [ ] index_add
- [ ] index_copy
- [ ] index_fill_.int_Scalar
- [ ] index_fill_.int_Tensor
- [ ] index_put_
- [ ] index_select
- [x] inverse
  - Status: Blocked
  - PR: https://github.com/pytorch/xla/pull/3575
  - Owner: @miladm 
- [ ] isnan
- [ ] kl_div
- [ ] kl_div_backward
- [ ] kthvalue
- [ ] l1_loss
- [ ] l1_loss_backward
- [ ] le.Scalar
- [ ] le.Tensor
- [ ] leaky_relu
- [ ] leaky_relu_backward
- [ ] lerp.Scalar
- [ ] lerp.Tensor
- [ ] linspace
- [x] log
  - Status: In Progress
  - PR: https://github.com/pytorch/xla/pull/3573
  - Owner: @miladm 
- [ ] log1p
- [x] log2
  - Status: In Progress
  - PR: https://github.com/pytorch/xla/pull/3573
  - Owner: @miladm 
- [x] log10
  - Status: In Progress
  - PR: https://github.com/pytorch/xla/pull/3573
  - Owner: @miladm 
- [ ] log_sigmoid_backward
- [ ] log_sigmoid_forward
- [x] logdet
  - Status: Blocked
  - PR: https://github.com/pytorch/xla/pull/3576
  - Owner: @miladm 
- [ ] logical_and
  - Status: In progress
  - PR: TBD
  - Owner: @wonjoolee95  
- [ ] logical_not
  - Status: In progress
  - PR: TBD
  - Owner: @wonjoolee95  
- [ ] logical_or
  - Status: In progress
  - PR: TBD
  - Owner: @wonjoolee95  
- [ ] logical_xor
  - Status: In progress
  - PR: TBD
  - Owner: @wonjoolee95  
- [ ] logsumexp
- [ ] lt.Scalar
- [ ] lt.Tensor
- [ ] masked_fill_.Scalar
- [ ] masked_fill_.Tensor
- [ ] masked_scatter_
- [ ] masked_select
- [ ] max
- [ ] max.dim
- [ ] max.dim_max
- [ ] max_pool2d_with_indices
- [ ] max_pool2d_with_indices_backward
- [ ] max_pool3d_with_indices
- [ ] max_pool3d_with_indices_backward
- [ ] max_unpool2d
- [ ] max_unpool3d
- [x] maximum
  - Status: Merged
  - PR: https://github.com/pytorch/xla/pull/3544
  - Owner: @JackCaoG 
- [ ] mean
- [ ] mean.dim
- [ ] min
- [ ] min.dim
- [ ] min.dim_min
- [ ] minimum
- [ ] mish
- [ ] mm
- [ ] mse_loss
- [ ] mse_loss_backward
- [ ] mul.Scalar
- [ ] mul.Tensor
- [ ] mv
- [ ] mv.out
- [ ] nan_to_num
- [ ] native_batch_norm
- [ ] native_batch_norm_backward
- [ ] ne.Scalar
- [ ] ne.Tensor
- [ ] neg
- [ ] nll_loss2d_backward
- [ ] nll_loss2d_forward
- [ ] nll_loss_backward
- [ ] nll_loss_forward
- [ ] nonzero
- [ ] norm.Scalar
- [ ] norm.ScalarOpt_dim
- [ ] norm.ScalarOpt_dim_dtype
- [ ] norm.ScalarOpt_dtype
- [ ] normal.float_Tensor
- [ ] normal.Tensor_float
- [ ] normal.Tensor_Tensor
- [ ] normal_
- [ ] permute
- [ ] pow.Scalar
- [ ] pow.Tensor_Scalar
- [ ] pow.Tensor_Tensor
- [ ] prelu
- [ ] prod
- [ ] prod.dim_int
- [ ] put_
- [ ] qr
- [ ] random_
- [ ] random_.from
- [ ] random_.to
- [ ] reciprocal
- [ ] reflection_pad2d
- [ ] reflection_pad2d_backward
- [ ] relu
- [ ] relu_
- [ ] remainder.Scalar
- [ ] remainder.Tensor
- [ ] repeat
- [ ] replication_pad1d
- [ ] replication_pad1d_backward
- [ ] replication_pad2d
- [ ] replication_pad2d_backward
- [ ] resize_
- [ ] roll
- [ ] round
- [ ] rrelu_with_noise
- [ ] rrelu_with_noise_backward
- [ ] rsqrt
- [ ] rsub.Scalar
- [ ] rsub.Tensor
- [ ] scatter.reduce
- [ ] scatter.src
- [ ] scatter.value
- [ ] scatter.value_reduce
- [ ] scatter_add
- [ ] select.int
- [ ] selu
- [ ] selu_
- [x] sgn
  - Status: Merged
  - PR: https://github.com/pytorch/xla/pull/3577
  - Owner: @miladm 
- [ ] sigmoid
- [ ] sigmoid_backward
- [x] sign
  - Status: Merged
  - PR: https://github.com/pytorch/xla/pull/3577
  - Owner: @miladm 
- [ ] silu.out
- [ ] silu_backward
- [ ] sin
  - Status: In progress
  - PR: https://github.com/pytorch/xla/pull/3581
  - Owner: @wonjoolee95  
- [ ] sinh
  - Status: In progress
  - PR: https://github.com/pytorch/xla/pull/3581
  - Owner: @wonjoolee95  
- [ ] slice.Tensor
- [x] slogdet
  - Status: Blocked
  - PR: https://github.com/pytorch/xla/pull/3576
  - Owner: @miladm 
- [ ] smooth_l1_loss
- [ ] smooth_l1_loss_backward
- [ ] softplus
- [ ] softplus_backward
- [ ] softshrink
- [ ] softshrink_backward
- [ ] sort
- [ ] sort.stable
- [ ] split.Tensor
- [ ] split_with_sizes
- [ ] sqrt
- [ ] squeeze
- [ ] squeeze.dim
- [ ] squeeze_
- [ ] squeeze_.dim
- [ ] stack
- [ ] std
- [ ] std.correction
- [ ] std.dim
- [ ] std_mean.correction
- [ ] sub.Scalar
- [ ] sub.Tensor
- [ ] sum
- [ ] sum.dim_IntList
- [ ] svd
- [ ] symeig
- [ ] t
- [ ] t_
- [ ] take
- [ ] tan
  - Status: In progress
  - PR: https://github.com/pytorch/xla/pull/3581
  - Owner: @wonjoolee95  
- [ ] tanh
- [ ] tanh_backward
- [ ] threshold
- [ ] threshold_backward
- [ ] topk
- [ ] trace
- [ ] transpose.int
- [ ] transpose_
- [ ] triangular_solve
- [ ] tril
- [ ] tril_
- [ ] triu
- [ ] triu_
- [ ] trunc
- [ ] unbind.int
- [ ] uniform_
- [ ] unsqueeze
- [ ] unsqueeze_
- [ ] upsample_bilinear2d
- [ ] upsample_bilinear2d_backward
- [ ] upsample_nearest2d
- [ ] upsample_nearest2d.vec
- [ ] upsample_nearest2d_backward
- [ ] upsample_nearest2d_backward.vec
- [ ] var.correction
- [ ] var_mean.correction
- [ ] view
- [ ] where.self
- [ ] xlogy.Tensor
- [ ] zero_

## Won't do's

This contains a list of ops that are not suitable for full codegen at the moment. Please list the op and the corresponding reason.
